### PR TITLE
Fix System.Collections.* XML comment errors

### PR DESCRIPTION
**Pull request by stephentoub**
These are causing errors in the source.dot.net build.

cc: @safern, @danmosemsft

### CODE REVIEW

File: src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs [hunk 1]
```diff
@@ -721,6 +721,7 @@ internal bool IsEmpty
             /// Add new item to the tail of the queue.
             /// </summary>
             /// <param name="item">The item to add.</param>
+            /// <param name="emptyToNonEmptyListTransitionCount"></param>
```
Comment (sharwell) [inline src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs]: 💡 Consider disabling CS1573 (missing parameter doc) and CS1712 (missing type parameter doc)

File: src/System.Collections.Concurrent/src/System/Collections/Concurrent/PartitionerStatic.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ namespace System.Collections.Concurrent
     /// <summary>
     /// Out-of-the-box partitioners are created with a set of default behaviors.  
     /// For example, by default, some form of buffering and chunking will be employed to achieve 
-    /// optimal performance in the common scenario where an IEnumerable<> implementation is fast and 
+    /// optimal performance in the common scenario where an IEnumerable{T} implementation is fast and
```
Comment (sharwell) [inline src/System.Collections.Concurrent/src/System/Collections/Concurrent/PartitionerStatic.cs]: 💡 May as well make this a reference?

```suggestion
    /// optimal performance in the common scenario where an <see cref="IEnumerable{T}"/> implementation is fast and 
```

File: src/System.Collections/src/System/Collections/Generic/HashSet.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@ namespace System.Collections.Generic
 {
     /// <summary>
     /// Implementation notes:
-    /// This uses an array-based implementation similar to Dictionary<T>, using a buckets array
+    /// This uses an array-based implementation similar to Dictionary{T}, using a buckets array
```
Comment (sharwell) [inline src/System.Collections/src/System/Collections/Generic/HashSet.cs]: 💡 

```suggestion
    /// This uses an array-based implementation similar to <see cref="Dictionary{TKey, TValue}"/>, using a buckets array
```

File: src/System.Collections/src/System/Collections/Generic/HashSet.cs [hunk 4]
```diff
@@ -218,7 +218,7 @@ public HashSet(int capacity, IEqualityComparer<T> comparer)
         #region ICollection<T> methods
 
         /// <summary>
-        /// Add item to this hashset. This is the explicit implementation of the ICollection<T>
+        /// Add item to this hashset. This is the explicit implementation of the ICollection{T}
```
Comment (sharwell) [inline src/System.Collections/src/System/Collections/Generic/HashSet.cs]: :bulb:

```suggestion
        /// Add item to this hashset. This is the explicit implementation of the <see cref="ICollection{T}"/>
```

File: src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs]: > Consider disabling CS1573 (missing parameter doc) and CS1712 (missing type parameter doc)

I have.  I just noticed this one and added it figuring at some point we'll enable those.  These don't actually break source.dot.net... just the malformed XML ones.

**Pull request merged**